### PR TITLE
fix: Fixed Attachment Metadata Being Set Incorrectly In Interaction Responses

### PR DESCRIPTION
**Pull request by Icebluewolf**
## Summary
In interaction responses attachment metadata was being placed in the wrong spot in the request body causing discord to ignore it.

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by plun1331 — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ These changes are available on the `master` branch, but have not yet been releas
   apps. ([#2650](https://github.com/Pycord-Development/pycord/pull/2650))
 - Fixed type annotations of cached properties.
   ([#2635](https://github.com/Pycord-Development/pycord/issues/2635))
+- Fixed attachment metadata being set incorrectly in interaction responses causing it to
+  be ignored. ([#2679](https://github.com/Pycord-Development/pycord/pull/2679))
```
Comment (JustaSqu1d) [inline CHANGELOG.md:68]:
```suggestion
- Fixed attachment metadata being set incorrectly in interaction responses which caused it to
  be ignored. ([#2679](https://github.com/Pycord-Development/pycord/pull/2679))
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]
slightly ambiguous wording fix

**Review by Dorukyum — APPROVED** [review]
The wording is fine

**Comment by JustaSqu1d** [comment]
> The wording is fine

It could be interpreted as the "interaction responses" being ignored as opposed to the "attachment metadata"

**Comment by Dorukyum** [comment]
> > The wording is fine
> 
> 
> 
> It could be interpreted as the "interaction responses" being ignored as opposed to the "attachment metadata"

I hadn't realised that but your suggestion wouldn't solve it

File: CHANGELOG.md (hunk 1 above)
Comment (Icebluewolf) [reply CHANGELOG.md]: Fixed with more clear wording of my own

**Comment by Icebluewolf** [comment]
Don't merge yet

File: discord/webhook/async_.py [hunk 2]
```diff
@@ -522,7 +522,7 @@ def create_interaction_response(
                     "content_type": "application/octet-stream",
                 }
             )
-        payload["attachments"] = attachments
+        payload["data"]["attachments"] = attachments
```
Comment (Dorukyum) [inline discord/webhook/async_.py:525]: Potential `KeyError` as shown in #2682.

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Comment by Icebluewolf** [comment]
Should Be Fixed
I tested
- Defer + Message
- Defer + Message With Attachment
- Message
- Message With Attachment

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]
that looks better lmao

**Pull request merged**